src/test/scala/cromwell/util/SampleWdl.scala

```diff
@@ -1668,4 +1668,32 @@ object SampleWdl {
       """.stripMargin
     override val rawInputs = Map.empty[String, String]
   }
+
+  /**
+    * Outputs which reference themselves.
+    */
+  object InputsAndOutputsAutoReferentiation extends SampleWdl {
```

Comment:
Ordering of inputs and outputs is definitely important - otherwise the motivating example for this change wouldn't work:
```
Int a = 5
Int b = a * 2
```